### ISSUE TITLE
Unsupported MPI flavor

### ISSUE THREAD

**Issue by ybarlev** [sdsc/fftw-roll#1]
Most test fail. "unsupported MPI flavor". Default compilation was used: "openmpi + gnu".

```
==============================
/root/rolltests/fftw.t 
ok 1 - fftw latest/gnu/rocks-openmpi installed
ok 2 - fftw v2.1.5/gnu/rocks-openmpi installed
ok 3 - module file for fftw/latest/gnu installed
No supported MPI flavor found
rollfftw.c:2:19: fatal error: fftw3.h: No such file or directory
 #include <fftw3.h>
                   ^
compilation terminated.
not ok 4 - compile/link using fftw/latest/gnu/rocks-openmpi
#   Failed test 'compile/link using fftw/latest/gnu/rocks-openmpi'
#   at /root/rolltests/fftw.t line 138.
No supported MPI flavor found
sh: ./rollfftw.gnu.exe: No such file or directory
not ok 5 - run using fftw/latest/gnu/rocks-openmpi double precision
#   Failed test 'run using fftw/latest/gnu/rocks-openmpi double precision'
#   at /root/rolltests/fftw.t line 141.
#                   ''
#     doesn't match '(?-xism:{ {6.00, 0.00} {-2.00, 2.00} {-2.00, 0.00} {-2.00, -2.00} })'
No supported MPI flavor found
rollfftw.c:2:19: fatal error: fftw3.h: No such file or directory
 #include <fftw3.h>
                   ^
compilation terminated.
not ok 6 - compile/link using fftw/latest/gnu/rocks-openmpi
#   Failed test 'compile/link using fftw/latest/gnu/rocks-openmpi'
#   at /root/rolltests/fftw.t line 144.
No supported MPI flavor found
sh: ./rollfftw.gnu.exe: No such file or directory
not ok 7 - run using fftw/latest/gnu/rocks-openmpi single precision
#   Failed test 'run using fftw/latest/gnu/rocks-openmpi single precision'
#   at /root/rolltests/fftw.t line 147.
#                   ''
#     doesn't match '(?-xism:{ {6.00, 0.00} {-2.00, 2.00} {-2.00, 0.00} {-2.00, -2.00} })'
ok 8 - module file for fftw/2.1.5/gnu installed
No supported MPI flavor found
rollfftw.c:6:18: fatal error: fftw.h: No such file or directory
 #include <fftw.h>
                  ^
compilation terminated.
not ok 9 - compile/link using fftw/2.1.5/gnu/rocks-openmpi
#   Failed test 'compile/link using fftw/2.1.5/gnu/rocks-openmpi'
#   at /root/rolltests/fftw.t line 163.
No supported MPI flavor found
sh: ./rollfftw.gnu.exe: No such file or directory
not ok 10 - run using fftw/2.1.5/gnu/rocks-openmpi
#   Failed test 'run using fftw/2.1.5/gnu/rocks-openmpi'
#   at /root/rolltests/fftw.t line 166.
#                   ''
#     doesn't match '(?-xism:{ {6.00, 0.00} {-2.00, 2.00} {-2.00, 0.00} {-2.00, -2.00} })'
1..10
# Looks like you failed 6 tests of 10.
[root@r-rocks ~]# module load rocks-openmpi
[root@r-rocks ~]# /root/rolltests/fftw.t 
ok 1 - fftw latest/gnu/rocks-openmpi installed
ok 2 - fftw v2.1.5/gnu/rocks-openmpi installed
ok 3 - module file for fftw/latest/gnu installed
No supported MPI flavor found
rollfftw.c:2:19: fatal error: fftw3.h: No such file or directory
 #include <fftw3.h>
                   ^
compilation terminated.
not ok 4 - compile/link using fftw/latest/gnu/rocks-openmpi
#   Failed test 'compile/link using fftw/latest/gnu/rocks-openmpi'
#   at /root/rolltests/fftw.t line 138.
No supported MPI flavor found
sh: ./rollfftw.gnu.exe: No such file or directory
not ok 5 - run using fftw/latest/gnu/rocks-openmpi double precision
#   Failed test 'run using fftw/latest/gnu/rocks-openmpi double precision'
#   at /root/rolltests/fftw.t line 141.
#                   ''
#     doesn't match '(?-xism:{ {6.00, 0.00} {-2.00, 2.00} {-2.00, 0.00} {-2.00, -2.00} })'
No supported MPI flavor found
rollfftw.c:2:19: fatal error: fftw3.h: No such file or directory
 #include <fftw3.h>
                   ^
compilation terminated.
not ok 6 - compile/link using fftw/latest/gnu/rocks-openmpi
#   Failed test 'compile/link using fftw/latest/gnu/rocks-openmpi'
#   at /root/rolltests/fftw.t line 144.
No supported MPI flavor found
sh: ./rollfftw.gnu.exe: No such file or directory
not ok 7 - run using fftw/latest/gnu/rocks-openmpi single precision
#   Failed test 'run using fftw/latest/gnu/rocks-openmpi single precision'
#   at /root/rolltests/fftw.t line 147.
#                   ''
#     doesn't match '(?-xism:{ {6.00, 0.00} {-2.00, 2.00} {-2.00, 0.00} {-2.00, -2.00} })'
ok 8 - module file for fftw/2.1.5/gnu installed
No supported MPI flavor found
rollfftw.c:6:18: fatal error: fftw.h: No such file or directory
 #include <fftw.h>
                  ^
compilation terminated.
not ok 9 - compile/link using fftw/2.1.5/gnu/rocks-openmpi
#   Failed test 'compile/link using fftw/2.1.5/gnu/rocks-openmpi'
#   at /root/rolltests/fftw.t line 163.
No supported MPI flavor found
sh: ./rollfftw.gnu.exe: No such file or directory
not ok 10 - run using fftw/2.1.5/gnu/rocks-openmpi
#   Failed test 'run using fftw/2.1.5/gnu/rocks-openmpi'
#   at /root/rolltests/fftw.t line 166.
#                   ''
#     doesn't match '(?-xism:{ {6.00, 0.00} {-2.00, 2.00} {-2.00, 0.00} {-2.00, -2.00} })'
1..10
# Looks like you failed 6 tests of 10.
```

**Comment by jjhayes** [comment]
Thanks for the report.  Our clusters all have the mpi-roll installed with openmpi_ib/mvapich2_ib, so support for the default rocks-openmpi is almost completely untested.  Having said that, it should work, and I'll check into this failure.

Jim

**Comment by jjhayes** [comment]
Thanks again for reporting this.  It turns out to be due to a bug in our fftw modulefile--a bug that's repeated in other mpi-dependent rolls.  I'll be checking in fixes to all of them in the next several days.

**Comment by tcooper** [comment]
@ybarlev, hope you don't mind I edited your comment adding Markdown formatting to your code block to make it more readable.

**Comment by ybarlev** [comment]
sure. I should have done this myself.

**Comment by jjhayes** [comment]
Resolved by commit ae8e16e76d6d8b2c056292a9e9f7eb307f50ffec.
